src/app/components/data/shared/rendering-container-image-picker/rendering-container-image-picker.component.ts

```diff
@@ -0,0 +1,151 @@
+    ],
+})
+
+export class RenderingContainerImagePickerComponent implements ControlValueAccessor, OnChanges, OnDestroy {
+
+    public get appDisplay() {
+        return this.UpperCaseFirstChar(this.app);
+    }
+
+    public get renderEngineDisplay() {
+        return this.UpperCaseFirstChar(this.renderEngine);
+    }
+
+    public appVersionControl = new FormControl();
+    public rendererVersionControl = new FormControl();
+
+    public appVersions: string[];
+    public containerImages: RenderingContainerImage[];
+
+    public containerImagesData: Observable<RenderingContainerImage[]>;
+    public containerImage: string;
+
+    @Input() public app: RenderApplication;
+    @Input() public imageReferenceId: string;
+    @Input() public renderEngine: RenderEngine;
+
+    private _app = new BehaviorSubject(null);
+    private _imageReferenceId = new BehaviorSubject(null);
+    private _renderEngine = new BehaviorSubject(null);
+
+    private _propagateChange: (value: string) => void = null;
+
+    private _destroy = new Subject();
+
+    constructor(
+        private changeDetector: ChangeDetectorRef,
+        private renderingContainerImageService: RenderingContainerImageService) {
+
+        combineLatest(this._app, this._imageReferenceId).pipe(
+            takeUntil(this._destroy),
+            switchMap(([app, imageReferenceId]) => {
+                return this.renderingContainerImageService.getAppVersionDisplayList(app, imageReferenceId);
+            }),
+        ).subscribe((appVersions) => {
+            this.appVersions = appVersions;
+            this.changeDetector.markForCheck();
+        });
+
+        this.rendererVersionControl.valueChanges.pipe(takeUntil(this._destroy)).subscribe((containerImage: string) => {
+            if (this._propagateChange) {
+                this._propagateChange(containerImage);
+            }
+            this.containerImage = containerImage;
+            this.changeDetector.markForCheck();
+        });
+
+        combineLatest(this._renderEngine, this._app, this._imageReferenceId, this.appVersionControl.valueChanges).pipe(
+            takeUntil(this._destroy),
+            switchMap(([renderEngine, app, imageReferenceId, value]) => {
+                return this.renderingContainerImageService.getContainerImagesForAppVersion(
+                    app, renderEngine, imageReferenceId, value);
+            }),
+        ).subscribe((containerImages: RenderingContainerImage[]) => {
+            this.containerImages = containerImages;
+            this.changeDetector.markForCheck();
+        });
+    }
+
+    public trackContainerImage(_, image: RenderingContainerImage) {
+        return image.containerImage;
+    }
+
+    public trackAppVersion(_, image: string) {
+        return image;
+    }
+
+    public writeValue(containerImageId: string) {
+        if (containerImageId) {
+            this.renderingContainerImageService.findContainerImageById(containerImageId)
+                .pipe(map(image => {
```

Comment:
here this is not going to get trigger you want to use `.subscribe`

Reply:
Updated - new behavior is almost as desired except rendererVersionControl label is not being updated when the value is